CMakeLists.txt

```diff
@@ -0,0 +1,98 @@
+cmake_minimum_required(VERSION 3.15.7)
+
+set(CMAKE_DISABLE_SOURCE_CHANGES ON)
+set(CMAKE_DISABLE_IN_SOURCE_BUILD ON)
+
+include(toolchain.cmake) # TODO(melvinw): this should not be included here. It should be configured by ide or by passing it to cmake
+
+# Toolchain must be included before project declaration
+project(MarsRoverFirmware)
+
+# Global Settings
+set(COMPILE_FLAGS
+        -MMD
+        -MP
+
+        -O3
```

Comment:
Is this easy to override?

Reply:
the O3? um depends what u mean by override. like u could just change that line lol.

Reply:
on a build by build basis, I believe u can append global flags in the cmd line and the later flags always take precedence